### PR TITLE
chore(helpers): change to use typescript

### PR DESCRIPTION
**Pull request by qiweiii**
#525

### CODE REVIEW

**Comment by qiweiii** [comment]
Build failed because I changed `"module": "ES2015"` to `"module": "commonjs"` in `tsconfig.json`.

I tried to follow https://github.com/TypeStrong/ts-node/issues/1007 to fix it, but not working yet, may need some time to figure it out

**Comment by imbhargav5** [comment]
Take your time . Good work @qiweiii

File: tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "experimentalDecorators": true,
     "moduleResolution": "node",
     "strictNullChecks": true, // enable strict null checks as a best practice
-    "module": "ES2015", // specify module code generation
+    "module": "commonjs", // specify module code generation
```
Comment (imbhargav5) [inline tsconfig.json:6]: Curious as to why this change is needed. @qiweiii .

File: rollup.config.js [hunk 2]
```diff
@@ -1,21 +1,23 @@
-import nodeResolve from '@rollup/plugin-node-resolve';
-import replace from '@rollup/plugin-replace';
-import commonjs from '@rollup/plugin-commonjs';
-import babel from '@rollup/plugin-babel';
-import json from '@rollup/plugin-json';
-import typescript2 from 'rollup-plugin-typescript2';
-import { terser } from 'rollup-plugin-terser';
-import sourceMaps from 'rollup-plugin-sourcemaps';
-import pkg from './package.json';
+import nodeResolve from "@rollup/plugin-node-resolve";
```
Comment (imbhargav5) [inline rollup.config.js:1]: Should this be a ts file or a js file? Not sure. Just wondering.

File: tsconfig.json (hunk 1 above)
Comment (qiweiii) [reply tsconfig.json:6]: It's because ts-node can only run our scripts when using 'commonjs', it's caused by a bug in ts-node(or maybe node): https://github.com/TypeStrong/ts-node/issues/935, and here is a related issue: https://github.com/TypeStrong/ts-node/issues/1007

File: rollup.config.js (hunk 2 above)
Comment (qiweiii) [reply rollup.config.js:1]: Do you mean this `rollup.config.js` file? I found they only started to support typescript config file in recent release: https://github.com/rollup/rollup/releases/tag/v2.52.0

Comment (imbhargav5) [reply rollup.config.js:1]: Yeah. Perhaps we can change it to `ts` too.

File: tsconfig.json (hunk 1 above)
Comment (imbhargav5) [reply tsconfig.json:6]: This might effect our build step.

Comment (imbhargav5) [reply tsconfig.json:6]: It will make our exported modules non-treeshakeable I believe.

Comment (imbhargav5) [reply tsconfig.json:6]: Never mind. This is good

File: helpers/create/index.ts [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+import fsExtra from "fs-extra";
+import path from "path";
+import replaceString from "replace-string";
+import makeDir from "make-dir";
+import ora from "ora";
+import execa from "execa";
+import truncate from "lodash.truncate";
+import addHookToListAndUpdate from "./addHookToListAndUpdate";
+
+export interface IHookDesc {
```
Comment (imbhargav5) [inline helpers/create/index.ts:12]: Let's move this to a separate file within helpers?

**Comment by imbhargav5** [comment]
This looks fantastic @qiweiii 🎉 🎉 🎉 🎉 

Did you QA them? Can you do one last iteration of QA and see if everything is working as expected. Then I will merge. 

Thanks for this!!

**Comment by qiweiii** [comment]
> This looks fantastic @qiweiii 🎉 🎉 🎉 🎉
> 
> Did you QA them? Can you do one last iteration of QA and see if everything is working as expected. Then I will merge.
> 
> Thanks for this!!

Yes, I have tested these scripts
